motoman_driver/src/simple_message/motoman_read_group_io_reply.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ * Copyright (c) 2021, Delft Robotics Institute
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *  * Neither the name of the Delft Robotics Institute, nor the names
+ *    of its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \author G.A. vd. Hoorn (TU Delft Robotics Institute)
+ */
+
+#include <string>
+#ifdef ROS
+#include "motoman_driver/simple_message/motoman_read_group_io.h"
+#include "motoman_driver/simple_message/motoman_read_group_io_reply.h"
+#include "simple_message/shared_types.h"
+#include "simple_message/log_wrapper.h"
+#endif
+
+#ifdef MOTOPLUS
+#include "motoman_read_group_io.h"         // NOLINT(build/include)
+#include "motoman_read_group_io_reply.h"   // NOLINT(build/include)
+#include "shared_types.h"                  // NOLINT(build/include)
+#include "log_wrapper.h"                   // NOLINT(build/include)
+#endif
+
+using industrial::shared_types::shared_int;
+namespace ReadGroupIOReplyResultCodes = motoman::simple_message::io_ctrl_reply::ReadGroupIOReplyResultCodes;
+
+namespace motoman
+{
+namespace simple_message
+{
+namespace io_ctrl_reply
+{
+
+ReadGroupIOReply::ReadGroupIOReply(void)
+{
+  this->init();
+}
+ReadGroupIOReply::~ReadGroupIOReply(void)
+{
+}
+
+void ReadGroupIOReply::init()
+{
+  // TODO( ): is '0' a good initial value?
+  this->init(0/*value*/, ReadGroupIOReplyResultCodes::SUCCESS);
```

Comment:
Any real concern here?

Reply:
well, I just don't know whether `0` is a correct default value.

I also wouldn't know any better value, or whether we'd want to use a sentinel of some sort here.

Technically, group IO is max 8 bit if I'm not mistaken, so there are certainly values outside the permissible range.